src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/WorkingDaysService.kt

```diff
@@ -7,29 +7,43 @@ import java.time.LocalDate
 @Service
 class WorkingDaysService(private val bankHolidayService: BankHolidayService) {
 
-  fun addWorkingDays(date: LocalDate, workingDays: Int): LocalDate {
-    val bankHolidays = bankHolidayService.getBankHolidaysForEnglandAndWales()
-    var adjustedDate = date
-    for (i in 0 until workingDays) {
-      adjustedDate = getNextWorkingDay(bankHolidays, adjustedDate)
-    }
-    return adjustedDate
+  fun workingDaysAfter(
+    date: LocalDate,
+  ): Sequence<LocalDate> {
+    return generateSequence(date) { it.nextWorkingDay() }.drop(1)
   }
 
-  fun isWeekend(date: LocalDate): Boolean {
-    val weekend = listOf(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY)
+  fun workingDaysBefore(
+    date: LocalDate,
+  ): Sequence<LocalDate> {
+    return generateSequence(date) { it.previousWorkingDay() }.drop(1)
+  }
+
+  fun isWeekend(date: LocalDate, weekend: List<DayOfWeek> = listOf(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY)): Boolean {
```

Comment:
if you don't parameterise `isNonWorkingDay` that would make with simpler too

Reply:
Done